app/helpers/maintenance_tasks/tasks_helper.rb

```diff
@@ -101,16 +101,33 @@ def csv_file_download_path(run)
       )
     end
 
-    # Return the appropriate field tag for the parameter, based on its type.
-    # If the parameter has a `validates_inclusion_of` validator, return a dropdown list of options instead.
-    def parameter_field(form_builder, parameter_name)
-      inclusion_validator = form_builder.object.class.validators_on(parameter_name).find do |validator|
+    # Returns the appropriate value for validator's inclusion options for a Proc, lambda, symbol, or other types.
+    #
+    # @param object [Task] The object for which the value needs to be resolved.
+    # @param parameter_name [String] The parameter name.
+    #
+    # @return [Object] value of the resolved inclusion option.
+    def resolve_inclusion_value(object, parameter_name)
+      inclusion_validator = object.class.validators_on(parameter_name).find do |validator|
         validator.kind == :inclusion
       end
+      return unless inclusion_validator
+
+      in_option = inclusion_validator.options[:in]
 
+      return in_option.call(object) if in_option.respond_to?(:call)
+      return object.send(in_option) if in_option.respond_to?(:to_sym)
```

Comment:
> I think we should support Enumerable except Ranges

How do you feel about symbols? I would like to keep the symbol support here _if possible_, simply because:
1. I really like the dropdown UI over free text
2. But I would like to be able to make an API call to get these values, ie.
```ruby
validates_inclusion_of :enum_attr, in: :enum_array, allow_nil: true

def enum_array
   # make an API call here
end
```

Reply:
The problem is actually doing network calls in the context of an HTTP request.

As it is, it's not super great that we may be doing this during the Run creation, in the POST maintenance_tasks/runs#create request. Removing that would mean a worse user experience (we couldn't show the validation errors if the validation ran in the job, and we would save a run just to show the errors), so we've decided that we wouldn't complicate validations and only run them on creation (we could have had two contexts of validations, one of them would be running before the run starts, in the job, allowing more expensive validations).

But using the same validations during the GET request to get to the Task page is not something I want. In particular with the refresh it means that your API call would happen every three seconds while a Run is running.

Reply:
That makes sense to me, I forgot that there's a refresh. I'm not sure if memoizing the results would prevent the repeated network calls. But does that mean we want to explicitly throw an error saying we don't support this? Because the approach I'm doing to fix the proc bug was to simply not render the dropdown, and let it flow to the default textarea, if we do the same for the symbol:

```ruby
return if in_option.respond_to?(:call) || in_option.respond_to?(:to_sym)
```

There's nothing stopping the user from still doing the code as above, the validation is still there, just the results are prettified into a dropdown form.